application/server/server_smoke_test.go

```diff
@@ -605,23 +611,67 @@ func checkForScanParams(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder,
 		}
 		return false
 	}, 10*time.Second, 10*time.Millisecond)
-	return scanParams
 }
 
-func checkAutofixDiffs(t *testing.T, c *config.Config, snykCodeScanParams types.SnykScanParams, loc server.Local) {
+func getIssueListFromPublishDiagnosticsNotification(t *testing.T, jsonRPCRecorder *testutil.JsonRPCRecorder, p product.Product, folderPath string) []types.ScanIssue {
+	t.Helper()
+
+	var issueList []types.ScanIssue
+	notifications := jsonRPCRecorder.FindNotificationsByMethod("textDocument/publishDiagnostics")
+	for _, n := range notifications {
+		diagnosticsParams := types.PublishDiagnosticsParams{}
+		_ = n.UnmarshalParams(&diagnosticsParams)
+		for _, diagnostic := range diagnosticsParams.Diagnostics {
+			diagnosticCode, ok := diagnostic.Code.(string)
+			if ok && diagnosticCode == "Snyk Error" {
+				continue
+			}
+			if diagnostic.Source != string(p) || filepath.Dir(uri.PathFromUri(diagnosticsParams.URI)) != folderPath {
+				continue
+			}
+
+			var issue types.ScanIssue
+			err := mapSerializedIssueToStruct(diagnostic.Data, &issue)
+
+			if err != nil {
+				continue
+			}
+			issueList = append(issueList, issue)
+		}
+	}
+
+	return issueList
+}
+
+func mapSerializedIssueToStruct(m any, result *types.ScanIssue) error {
```

Comment:
ah, data is a map[string]string as it's LSPAny type?

Reply:
Or even define the data attribute directly as types.ScanIssues to avoid this serialization dance.

Reply:
I think it should be ok. We will only need that when we deserialize the the types.*Params. 
Which we don't do except in tests afaik

Reply:
I like that type.go maps the protocol def 1:1. Makes it easier to see and compare